libdnf/rpm/solv_query.cpp

```diff
@@ -1790,10 +1790,11 @@ SolvQuery & SolvQuery::ifilter_installed() {
         return *this;
     }
     solv::SolvMap filter_result(p_impl->sack->pImpl->get_nsolvables());
-    for (PackageId candidate_id : p_impl->query_result) {
-        Solvable * solvable = solv::get_solvable(pool, candidate_id);
+    for (auto candidate_id = installed_repo->start; candidate_id < installed_repo->end; ++candidate_id) {
+        auto pkg_candidate_id = PackageId(candidate_id);
```

Comment:
I suggest that it is not right approach, but definitely good direction. Between `installed_repo->start` and `candidate_id < installed_repo->end` there could be Ids that are not packages. In theory it should be not harmful.
In case that `p_impl->query_result` is empty or with few elements, the new logic will be slower.
What about to iterate over `p_impl->query_result` with interval between `installed_repo->start`; `candidate_id < installed_repo->end`?